ci/scenario_helpers/data_writing.py

```diff
@@ -29,100 +28,172 @@ class DataWriter(object):
-                   screen=True, file_locations=None, ee_info=None, logger=LOGGER):
+    def write_data_fio(cls, client, fio_configuration, edge_configuration=None, file_locations=None, fio_vdisk_limit=FIO_VDISK_LIMIT,
+                       screen=True, loop_screen=True, logger=LOGGER):
         """
-        Write data to a specific host.
-        Can write data within a screen
-        :param client: ovs ssh client for the vm
-        :type client: ovs.extensions.generic.sshclient.SSHClient
-        :param cmd_type: type of command. Was used to differentiate between dd and fio
-        :type cmd_type: str
-        :param configuration: configuration params for fio.First value represents read, second one write percentage eg (10, 90)
-        :type configuration: tuple
-        :param data_to_write: amount of data to write eg 10 * 1024 ** 3
-        :type data_to_write: int
-        :param fio_vdisk_limit: amount of vdisks to handle with one fio instance. Defaults to 50
-        :type fio_vdisk_limit: int
+        Start writing data using fio
+        Will output to files within /tmp/
+        :param client: 
+        :param fio_configuration: configuration for fio. Specify iodepth and bs
+        :type fio_configuration: dict {'bs': '4k', 'iodepth': 32}
         :param edge_configuration: configuration to fio over edge -OPTIONAL eg {'port': 26203, 'hostname': 10.100.10.100, 'protocol': tcp|udp, 'fio_bin_location': /tmp/fio.bin, 'volumename': ['myvdisk00']}
         :type edge_configuration: dict
-        :param screen: offload to screen
-        :type screen: bool
         :param file_locations: in conjunction with edge_configuration=None, points towards the files to perform fio on -OPTIONAL
         :type file_locations: list
-        :param ee_info: settings for the entreprise edition - OPTIONAL eg {'username': 'test', 'password': 'test'}
-        :type ee_info: dict
+        :param fio_vdisk_limit: amount of vdisks to handle with one fio instance. Defaults to 50
+        :type fio_vdisk_limit: int
+        :param screen: Offload to screen. Defaults to True
+        :type screen: bool
+        :param loop_screen: Keep looping the command in the screen. Defaults to True
+        :type loop_screen: bool
         :param logger: logging instance
-        :type logger: ovs.log.log_handler.LogHandler
-        :raises OSError: if it fails to create a output directory
-        :raises ValueError: if an unknown cmd_type is supplied
-        :return: list of screen names (empty if screen is False), list of output files
-        :rtype: tuple(list, list)
+        :return: 
         """
-        bs = '4k'
-        iodepth = 32
-        fio_output_format = 'json'
-        write_size = data_to_write
-        cmds = []
+        if edge_configuration is None and file_locations is None:
+            raise ValueError('Either edge configuration or file_locations need to be specified')
```

Comment:
These variables are opposites to one another. If the edge_configuration is passed, file locations isn't needed and vise versa but one of the two is required to configure the fio therefore I check on both of them.